pkg/injector/envoy_config.go

```diff
@@ -214,5 +278,37 @@ func getXdsCluster(config envoyBootstrapConfigMeta) map[string]interface{} {
 				},
 			},
 		},
+	}, nil
+}
+
+func xdsToYaml(m protoreflect.ProtoMessage) ([]byte, error) {
+	marshalOptions := protojson.MarshalOptions{
+		UseProtoNames: true,
+	}
+	configJSON, err := marshalOptions.Marshal(m)
+	if err != nil {
+		return nil, err
 	}
+
+	// Convert the JSON to an object.
+	// Reference impl taken from https://github.com/ghodss/yaml/blob/master/yaml.go#L87
+	var jsonObj interface{}
+	// We are using yaml.Unmarshal here (instead of json.Unmarshal) because the
+	// Go JSON library doesn't try to pick the right number type (int, float,
+	// etc.) when unmarshalling to interface{}, it just picks float64
+	// universally. go-yaml does go through the effort of picking the right
+	// number type, so we can preserve number type throughout this process.
+	err = yaml.Unmarshal([]byte(configJSON), &jsonObj)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This can be a separate function that references https://github.com/ghodss/yaml/blob/v1.0.0/yaml.go#L89 (references code for a release tag instead of master).

Reply:
Updated